src/metabase/lib/query.cljc

```diff
@@ -141,7 +182,8 @@
 
 (defmethod query-method :query ; legacy MBQL query
   [metadata-providerable legacy-query]
-  (query-from-legacy-query metadata-providerable legacy-query))
+  (query-from-legacy-query metadata-providerable
```

Comment:
if these calls were inverted `(add-types-to-fields (query-from-legacy-query ....))` would that make handling legacy fields unnecessary in `add-types-to-fields`? why not?

Reply:
Problem with that approach is that `query-from-legacy-query` calls `query-with-stages`. `query-with-stages` must return `::lib.schema/query` to conform schema. Hence the reversion of calls produces the error. That error was the other problem, present on master only.

To simplify `add-types-to-fields`, I've moved the call to it into `query-from-legacy-query` right after the query is transformed to pmbql and before `query-with-stages` call.

I've also added there binding of `*suppress-expression-type-check?*` to avoid warnings that come from `lib.convert/->pMBQL :dispatch-type/map`, specifically from normalization.

As the same var is already bound also in `clean-stage-schema-errors`, thanks to that previous avoids just warnings and not the clause missing effective type removal.

---

`add-types-to-fields` could then be simplified to handle pmbql only. However with that simplification in place I've started seeing lots of test failures. I suspected it is because base types are added now during the mbql -> pmbql transformation and are left in the queries during the conversion back to mbql. To mitigate that, I'm adding `:metabase.lib.query/transformation-added-base-type` key to options. If field options contain that, base type is removed during transformation back to legacy.